third_party/terraform/utils/config.go.erb

```diff
@@ -735,7 +739,15 @@ func (c *Config) GetCredentials(clientScopes []string) (googleoauth.Credentials,
 		if err != nil {
 			return googleoauth.Credentials{}, fmt.Errorf("error loading credentials: %s", err)
 		}
+		if c.ImpersonateServiceAccount != "" {
```

Comment:
Hi @upodroid! Sorry, I have been in discussions with @rileykarson the last couple of days on how we might want this to work. Given we need the `caller` to authenticate first, is it possible for the `caller` to authenticate with an `AccessToken`, `Credentials` or the default credentials? In which case we would want a similar block like this in the if-block above? It looks like it could potentially be possible by using `option.WithTokenSource`, but I haven't dug into it too much.

Reply:
I can do that but where is access token being provided from? I can only think of vault gcp access token, access_token datasource or someone manually generates one and loads it in to terraform which I highly doubt anyone is doing that.

Reply:
It added that in but i'm curious about the answer. IMHO it is not needed. Do you think we can get it merged intime for the next minor release?

Reply:
While I agree, it's not impossible for someone to be using it this way, and we'd prefer to be consistent and add this functionality in now than have to add it in later.  It won't be in for the next minor release, as that release has already been cut, but if tests pass now I can have it merged today and it would go in the next minor release after that.

Reply:
cool